qemu/tests/queues_number_test.py

```diff
@@ -0,0 +1,198 @@
+import logging
+import re
+import os
+
+from virttest import error_context
+from virttest import data_dir
+from virttest import utils_net
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import utils_netperf
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    MULTI_QUEUE chang queues number test
+    1) Boot up VM, and login guest
+    2) Enable the queues in guest
+    3) Run netperf_and_ping test
+    4) Change queues number repeatedly during netperf_and_ping stress testing
+    5) Reboot VM
+    6) Repeat above 1-4 steps
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def change_queues_number(ifname, q_number, queues_status=None):
+        """
+        change queues number
+        """
+        if not queues_status:
+            queues_status = get_queues_status(ifname)
+        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
+        status, output = session_serial.cmd_status_output(mq_set_cmd, safe=True)
+        if status:
+            test.fail("Failed to change queues number,"
+                      "status=%s, output=%s" % (status, output))
+        cur_queues_status = get_queues_status(ifname)
+
+        if q_number != queues_status[1] and queues_status[0] >= q_number > 0:
+            if (cur_queues_status[1] != q_number
+                    or cur_queues_status[0] != queues_status[0]):
+                logging.info("Param is valid, but change queues failed")
```

Comment:
Only log now? Shouldn't fail the test?

Reply:
I'm fine with the previous error message, which is very clear.

Reply:
ok, I will change it.